1 implement  main module

hisabimbola opened the pull request:
#### Summary
<!-- Provide a short summary of your changes -->

#### Description
<!-- Describe the changes in this PR here and provide some context -->

#### TODO

- Tests
    - [ ] Unit
    - [ ] Integration
    - [ ] Acceptance
- [ ] Documentation

hisabimbola committed:
* feat(lineitemstateupdate): add lineitemstate  submodule

* feat(lineitemstateupdate): add initial class module

* feat(lineitemstate): add parse and process method

Commit:
fix(CLI): chmod CLI file to be executable

hisabimbola committed:
Add returninfo submodule (#3)

* feat(lineitemstateupdate): add initial class module

* feat(lineitemstate): add parse and process method

* feat(add-returninfo): implement submodule

* refactor(return-info): make code better

* WIP

* refactor reduce some more

* flatten the if else statements

hisabimbola committed:
Implement cli (#6)

* feat(get-api-credentials): add method

* feat(CLI): add basic implementation

* feat(CLI): add returnInfo module to CLI

* chore(package.json): improve npm coverage script

Siilwyn commented on test/add-returninfo-csv-parser.spec.js:
```diff
@@ -0,0 +1,217 @@
+else
+  PROJECT_KEY = process.env.npm_config_projectkey
+
+const logger = {
```
Linter warning here, switching to `process.xxx` should work. Then it's also easy to find temporary debugging code that log stuff.

Siilwyn commented on src/lineitemstate-csv-parser.js:
```diff
@@ -0,0 +1,83 @@
+
+export default class LineItemStateCsvParser {
+  constructor (apiClientConfig, logger, config = {}) {
+    const selfPkg = loadSelfPkg.sync(process.cwd())
```
Since it's sync and the `package.json` is always in the root maybe we can just get the version with `import` so we don't need a module for it. Like [on this line](https://github.com/commercetools/csv-parser-orders/pull/7/files#diff-d9d9214113dea1c364abd4ba3c70be11R10).

Siilwyn reviewed (changes requested):
'main.spec.js' has interesting tests. ;)
'main.js' has interesting code too ;)

That said the "main" field in the package.json is empty and there is no "bin" field either.

Siilwyn committed:
style(variable): do not declare the parsed result as an additional variable

Commit:
fix(returninfo): remove invalid logger defaults

Logger default added was `process.stderr` and it's expected to be called like

`process.stderr()`

this will throw an error if called because there is no method like that

what is

present is `process.stderr.write()` because it's a stream. Two solutions is to bind or use

`console.log`.

I went with `console.log` ;)

Commit:
fix(lineitemstate): remove invalid logger defaults

the logger default added was `process.stderr` and it's expected to be called like

`process.stderr()` this will throw an error if called because there is no method like that

what

is present is `process.stderr.write()` because it's a stream. Two solutions is to bind or use

`console.log`.

 I went with `console.log` ;)

Commit:
chore(logging): add a start conversion info

Commit:
chore(CLI): update log file name

Commit:
refactor(userAgent): use sphere-node-utils module to build userAgent

Commit:
style(src): remove unused eslint disable command

Siilwyn commented on src/add-return-info-csv-parser.js:
```diff
@@ -29,7 +35,7 @@ export default class AddReturnInfoCsvParser {
 
   parse (input, output) {
-    this.logger.verbose('Starting conversion')
+    this.logger.info('Starting Return Info CSV conversion')
```
🎉

Commit:
docs(readme): update readme

Commit:
chore(package.json): add entry file and bin field

Commit:
test(cli): make tests 100% 🌞
